pkg/trace/info/stats.go

```diff
@@ -288,6 +284,51 @@ func (s *SpansMalformed) String() string {
 	return mapToString(s.tagValues())
 }
 
+// maxAbsPriority specifies the absolute maximum priority for stats purposes. For example, with a value
+// of 10, the range of priorities reported will be [-10, 10].
+const maxAbsPriority = 10
+
+// samplingPriorityStats holds the sampling priority metrics that will be reported every 10s by the agent.
+type samplingPriorityStats struct {
+	// counts holds counters for each priority in position maxAbsPriorityValue + priority.
+	// Priority values are expected to be in the range [-10, 10].
+	counts [maxAbsPriority*2 + 1]int64
+}
+
+// CountSamplingPriority increments the counter of observed traces with the given sampling priority by 1
+func (s *samplingPriorityStats) CountSamplingPriority(p sampler.SamplingPriority) {
+	if p > (-1*maxAbsPriority) && p < maxAbsPriority {
+		atomic.AddInt64(&s.counts[maxAbsPriority+p], 1)
+	}
+}
+
+// reset sets stats to 0
+func (s *samplingPriorityStats) reset() {
+	for i := 0; i < len(s.counts); i++ {
```

Comment:
```suggestion
	for i := range s.counts {
```
It's the same thing.

Reply:
Agreed